tests/unit/rdf/test_resource_serialisation.py

```diff
@@ -314,6 +315,130 @@ class A(NewResource):
+
+def test_uncommon_html_tag_detected_in_markdown(capsys):
+    """Testing to see if a uncommon, less frequently used, tag is detected in a markdown """
+    example_html_in_markdown = """
+    # This is some example markdown
+    <blockquote cite="http://www.worldwildlife.org/who/index.html">
+    For nearly 60 years, WWF has been protecting the future of nature. The world's leading conservation organization, WWF works in 100 countries and is supported by more than one million members in the United States and close to five million globally.
+    </blockquote>
+    """
+    map_str_to_markdown(example_html_in_markdown)
+    captured = capsys.readouterr()
+    assert captured.out == "True\n"
+
+def test_html_images_detected_in_markdown(capsys):
+    """Test that an image tag, with referece to an image, gets detected in markdown"""
+    example_html_in_markdown = """
+    # This is some example markdown
+    <img src="img_girl.jpg" alt="Girl with a jacket" width="500" height="600">
+    """
+    map_str_to_markdown(example_html_in_markdown)
+    captured = capsys.readouterr()
+    assert captured.out == "True\n"
+
+def test_mixture_of_html_tags_detected_in_markdown(capsys):
+    """Testing that a mixture of tags are detected in markdown """
+    example_html_in_markdown = """
+    <!DOCTYPE html>
+    <html>
+    <body>
+
+    <h2>HTML Buttons</h2>
+    <p>HTML buttons are defined with the button tag:</p>
+
+    <button>Click me</button>
+
+    </body>
+    </html>
+    """
+    map_str_to_markdown(example_html_in_markdown)
+    captured = capsys.readouterr()
+    assert captured.out == "True\n"
+
+def test_none_tag_detected_in_markdown(capsys):
+    """Testing variations of greater than and less than sign don't get identified as html"""
+    example_html_in_markdown = """
+    # This is some example markdown without any html
+    but does contain these < > and this <> 
+    """
+    map_str_to_markdown(example_html_in_markdown)
+    captured = capsys.readouterr()
+    assert captured.out == "False\n"
+
+def test_algebraic_expressions_with_spaces_detected_in_markdown(capsys):
+    """Testing algebraic expressions that have spaces between the 
+    greater than and less than symbols don't get identified as html"""
+    example_html_in_markdown = """
+    x > y,
+    x < y,
+    x < z < y,
+    x > y > z
+    """
+    map_str_to_markdown(example_html_in_markdown)
+    captured = capsys.readouterr()
+    assert captured.out == "False\n"
+
+def test_algebraic_expressions_without_spaces_detected_in_markdown(capsys):
+    """Testing algebraic expressions get identified as html even tho
+    they cleary aren't"""
+    example_html_in_markdown = """
+    x<z<y , x>y>z
+    """
+    map_str_to_markdown(example_html_in_markdown)
+    captured = capsys.readouterr()
+    assert captured.out == "True\n"
+
+def test_no_html_detected_in_markdown(capsys):
+    """Testing that no html is detected in a text only markdown"""
+    example_html_in_markdown = """
```

Comment:
I think we need to test for an extreme case here. For example, `example_html_in_markdown  = "html is Hypertext Markup Language. And it includes tags like <label>, <br> and <p>"`. Maybe this example is too extreme and we are confident that this will _almost_  never happen. @robons What do you think?

Reply:
Also, I heard some suggestion of testing that the parser works with all HTML tags. Since our parser is based off python's built in [HTMLParser](https://docs.python.org/3/library/html.parser.html), I think we can trust that they have tested it thoroughly enough.

Reply:
I think I see what you're saying now @GDonRanasinghe. For instance the user may define the following markdown:

```markdown
# My Document

This is some text and here is some literal text that isn't rendered as HTML: `<p>Some paragraph</p>`.
```

Now you're right, technically that isn't active HTML since it's escaped inside the backticks - so it'll be displayed as literal text. However, we're happy enough for the above example to falsely claim it contains active HTML at the moment since we don't want to go to the difficulty of writing a markdown parser ourselves.